src/lib.rs

```diff
@@ -0,0 +1,107 @@
+use std::env;
+
+use log::info;
+use tokio::sync::mpsc;
+
+use trin_core::cli::TrinConfig;
+use trin_core::jsonrpc::launch_jsonrpc_server;
+use trin_core::portalnet::protocol::{
+    HistoryNetworkEndpoint, JsonRpcHandler, PortalEndpoint, PortalnetConfig, PortalnetProtocol,
+    StateNetworkEndpoint,
+};
+use trin_history::HistoryRequestHandler;
+use trin_state::StateRequestHandler;
+
+pub async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    println!("Launching trin");
+
+    let trin_config = TrinConfig::new();
```

Comment:
My understanding is that we want to run the different networks on separate UDP ports. If that is the case, we need to extend the CLI parser (StructOpt) to support command line args for more than one network.

Reply:
> My understanding is that we want to run the different networks on separate UDP ports.

This doesn't sound right to me. The subnets would all be running off of the same discv5 layer, which is what has opened the UDP port. Then the overlay messages are all happening over talkreq over the shared discv5.

Reply:
Tracking this discussion [here](https://github.com/ethereum/trin/issues/90)